### PR TITLE
Ignore order of arguments while doing layernorm + pointwise fusion

### PR DESCRIPTION
**Pull request by umangyadav**
See the conversation here: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2185#discussion_r1325216555

### CODE REVIEW

**Review by TedThemistokleous — APPROVED** [review]

**Review by shivadbhavsar — APPROVED** [review]

File: test/gpu/fuse_ops.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#include <migraphx/gpu/fuse_ops.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/program.hpp>
+#include <basic_ops.hpp>
+#include <migraphx/make_op.hpp>
+#include <test.hpp>
+#include <pointwise.hpp>
+
+void run_pass(migraphx::program& p)
```
Comment (pfultz2) [inline test/gpu/fuse_ops.cpp:35]: You should run the pass on the module not the program.

File: test/gpu/fuse_ops.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+{
+    migraphx::run_passes(p, {migraphx::gpu::fuse_ops{}, migraphx::dead_code_elimination{}});
+}
+
+TEST_CASE(layernorm_pointwise)
+{
+    migraphx::shape s{migraphx::shape::float_type, {2, 3, 4}};
+    auto create_program = [=](bool first_arg_layernorm) {
+        migraphx::program p;
+        auto* mm       = p.get_main_module();
```
Comment (pfultz2) [inline test/gpu/fuse_ops.cpp:45]: Just create a module: `module mm;` and return that.

File: test/gpu/fuse_ops.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        auto alloc_ins3 = mm->add_instruction(alloc);
+        pw_inputs.push_back(alloc_ins3);
+        auto add2 = mm->add_instruction(make_precompile_op("pointwise"), pw_inputs, {pw_add2});
+        mm->add_return({add2});
+        return p;
+    };
+
+    auto create_fused_program = [=]() {
+        migraphx::program p;
+        auto* mm       = p.get_main_module();
```
Comment (pfultz2) [inline test/gpu/fuse_ops.cpp:74]: Same here this should just return the module.

File: test/gpu/fuse_ops.cpp (hunk 2 above)
Comment (pfultz2) [reply test/gpu/fuse_ops.cpp:45]: Nevermind you need the program to create the submodules.

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**